jscomp/others/js_string.ml

```diff
@@ -75,6 +75,7 @@ external repeat : int -> t = "" [@@bs.send.pipe: t] (** ES2015 *)
 
 external replace : t ->  t ->  t = "" [@@bs.send.pipe: t]
 external replaceByRe : Js_re.t ->  t ->  t = "replace" [@@bs.send.pipe: t]
+external replaceByFun1 : Js_re.t -> (t -> t -> t) -> t = "replace" [@@bs.send.pipe: t]
 
```

Comment:
hmm, yea this is a tricky one. In good js fashion it has varargs right in the middle of the argument list, and the number of args of course depend on the actual regex. I think just `replaceByN` should be descriptive enough, but I'm unsure if it should also be prefixed with `unsafe`.

This is also missing the offset and source arguments, though that would make the function even more unsafe.

How about this:
* Name the functions `unsafeReplaceByN`
* Add offset and source arguments
* Add a 0 submatch variant
* Add tests so we maintain full coverage
* It would be great if you could also add some doc comments. This module isn't a shining example of good documentation, but look at `Js_re` for what we're aiming for.

In the future I think pehaps we should build a safe abstraction around this function, with a callback that receives a match object so we can do some checks when it's accessed.